master/buildbot/test/unit/test_process_properties.py

```diff
@@ -1232,6 +1232,42 @@ def testFlattenListAdd2(self):
                       ["string", "bla", "string", "bla"])
         return d
 
+    @defer.inlineCallbacks
+    def testCompEq(self):
+        self.props.setProperty("do-tests", "string", "scheduler")
+        result = yield self.build.render(Property("do-tests") == "string")
+        self.assertEqual(result, True)
```

Comment:
Works, since Python uses reflection with the rhs comparators if the type on the lhs throws NotImplemented.